<?xml version="1.0"?>
<launch>
    <!-- <args for urdf/> -->
    <arg name="camera_model"         default="BW1098OAK" /> <!-- 'zed' or 'zedm' -->
    <arg name="camera_name"          default="oak" />

    <arg name="base_frame"           default="oak-d_frame" />

    <arg name="cam_pos_x"             default="0.0" />    <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_y"             default="0.0" />    <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_z"             default="0.0" />    <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_roll"              default="1.5708" /> <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_pitch"             default="0.0" />    <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_yaw"               default="1.5708" /> <!-- Orientation respect to base frame (i.e. "base_link) -->

    <arg name="camera_param_uri"      default="package://oak_interface/params/camera" />

 
    <include file="$(find oak_interface)/launch/urdf.launch">
        <arg name="base_frame"      value="$(arg  base_frame)"  />
        <arg name="camera_model"    value="$(arg  camera_model)"/>
        <arg name="camera_name"     value="$(arg  camera_name)" />
        <arg name="cam_pos_x"       value="$(arg  cam_pos_x)"   />
        <arg name="cam_pos_y"       value="$(arg  cam_pos_y)"   />
        <arg name="cam_pos_z"       value="$(arg  cam_pos_z)"   />
        <arg name="cam_roll"        value="$(arg  cam_roll)"    />
        <arg name="cam_pitch"       value="$(arg  cam_pitch)"   />
        <arg name="cam_yaw"         value="$(arg  cam_yaw)"     />
    </include>


    <!-- Cameras used -->
    <param name="publish_mono"      type="bool" value="false" />
    <param name="publish_depth"     type="bool" value="false" />
    <param name="publish_rectified" type="bool" value="false" />
    <param name="publish_rgb"       type="bool" value="false" />

    <!-- Cameras parameters -->
    <param name="mono_camera_resolution" type="int"    value="720"  /> <!-- 720, 800, 400 -->
    <param name="mono_camera_fps"        type="double" value="60.0" /> <!-- Max 120-->

    <param name="rgb_camera_resolution" type="int"    value="1080"  /> <!-- 1080, 3840 (4K), 2826 (12MP) -->
    <param name="rgb_camera_fps"        type="double" value="60.0" />  <!-- Max 60-->

    <!-- IMU -->
    <param name="publish_imu"       type="bool" value="true" />

    <!-- Neural Network -->
    <param name="publish_rgb_detections" type="bool" value="false" />
    <param name="publish_stereo_detections" type="bool" value="true" /> 
    <param name="nnBlobPath"         type="string" value="$(find oak_interface)/resources/mobilenet-ssd_openvino_2021.2_6shave.blob" />

    <!-- Node -->
    <node name="oak" pkg="oak_interface" type="oak" output="screen" required="true">
        <param name="camera_name" value="$(arg camera_name)"/>
        <param name="camera_param_uri" value="$(arg camera_param_uri)"/>

        <remap from="left/image"  to="/oak/left/image"/>
        <remap from="left/info"   to="/oak/left/info"/>
        <remap from="right/image" to="/oak/right/image"/>
        <remap from="right/info"  to="/oak/right/info"/>

        <remap from="left_rectified/image"  to="/oak/left_rectified/image"/>
        <remap from="left_rectified/info"   to="/oak/left_rectified/info"/>
        <remap from="right_rectified/image" to="/oak/right_rectified/image"/>
        <remap from="right_rectified/info"  to="/oak/right_rectified/info"/>

        <remap from="depth/image"  to="/oak/depth/image"/>
        <remap from="depth/info"   to="/oak/depth/info"/>

        <remap from="rgb/image"  to="/oak/rgb/image"/>
        <remap from="rgb/info"   to="/oak/rgb/info"/>

        <remap from="/detections_images"    to="/oak/detections/image"/>
        <remap from="/detections_depth"     to="/oak/detections/depth"/>
        <remap from="/detections_spatial"     to="/oak/detections/spatial"/>

        <remap from="Imu"        to="/oak/imu"/>
        
    </node>

    <node name = "rqt_dashboard" pkg = "rqt_gui" type = "rqt_gui" respawn = "false" output = "screen" args = "--perspective-file $(find oak_interface)/rqt_config/oak.perspective" /> 

    <node pkg="rqt_image_view" type="rqt_image_view" name="image_view_detections" args="/oak/detections/image"/>


    <!-- <node pkg="rqt_image_view" type="rqt_image_view" name="image_view_left"/>  -->

 <!--   <node type="rviz" name="rviz" pkg="rviz" args="-d $(find oak_interface)/rviz/oak.rviz" /> -->

 <!-- 
    <node pkg="rqt_image_view" type="rqt_image_view" name="image_view_left" args="/oak_interface_process/left/image"/>
    <node pkg="rqt_image_view" type="rqt_image_view" name="image_view_right" args="/oak_interface_process/right/image"/>
    <node pkg="rqt_image_view" type="rqt_image_view" name="image_view_left_rectified" args="/oak_interface_process/left/image_rectified"/>
    <node pkg="rqt_image_view" type="rqt_image_view" name="image_view_right_rectified" args="/oak_interface_process/right/image_rectified"/>
    <node pkg="rqt_image_view" type="rqt_image_view" name="image_view_depth" args="/oak_interface_process/stereo/depth"/>   -->   

</launch>
